pkg/pusher/pusher.go

```diff
@@ -190,6 +198,21 @@ LOOP:
 						err = fmt.Errorf("pusher: receipt storer address: %w", err)
 						return
 					}
+
+					po := swarm.Proximity(ch.Address().Bytes(), storerPeer.Bytes())
+					d := s.depther.NeighborhoodDepth()
+					if po < d {
+						mtx.Lock()
+						retryCounter[ch.Address().ByteString()]++
```

Comment:
increment after the check with the counter. Otherwise you will effectively only retry `retryCount-1`

Reply:
since we already retried, this is the correct thing to do. you can double check